pkg/build/controller/factory/factory.go

```diff
@@ -218,16 +218,22 @@ type typeBasedFactoryStrategy struct {
 }
 
 func (f *typeBasedFactoryStrategy) CreateBuildPod(build *buildapi.Build) (*kapi.Pod, error) {
+	var pod *kapi.Pod
+	var err error
 	switch build.Parameters.Strategy.Type {
 	case buildapi.DockerBuildStrategyType:
-		return f.DockerBuildStrategy.CreateBuildPod(build)
+		pod, err = f.DockerBuildStrategy.CreateBuildPod(build)
 	case buildapi.STIBuildStrategyType:
-		return f.STIBuildStrategy.CreateBuildPod(build)
+		pod, err = f.STIBuildStrategy.CreateBuildPod(build)
 	case buildapi.CustomBuildStrategyType:
-		return f.CustomBuildStrategy.CreateBuildPod(build)
+		pod, err = f.CustomBuildStrategy.CreateBuildPod(build)
 	default:
 		return nil, errors.New("No strategy defined for type")
 	}
+	if pod != nil {
+		pod.Annotations[buildapi.BuildAnnotation] = build.Name
```

Comment:
this annotation key is generic enough that I'd be nervous using it as a strong indicator that this pod is itself a build pod (I could see pods getting annotated with the build that spawned them, for example, at some point in the future)